src/stores/activities.ts

```diff
@@ -31,75 +32,109 @@ export const activityMetadataDefinitions = gqlSubscribable<ActivityMetadataDefin
+);
 
-              if (byTimelineId[timeline.id] === undefined) {
-                byTimelineId[timeline.id] = [activity];
-              } else {
-                byTimelineId[timeline.id].push(activity);
+export const activityDirectivesByView: Readable<ActivityDirectivesByView> = derived(
+  [activityDirectivesList, view],
+  ([$activityDirectivesList, $view]) => {
+    const { definition } = $view;
+    const { plan } = definition;
+    const { timelines } = plan;
+    const byLayerId: Record<number, ActivityDirective[]> = {};
+    const byTimelineId: Record<number, ActivityDirective[]> = {};
+
+    for (const activityDirective of $activityDirectivesList) {
+      for (const timeline of timelines) {
+        const { rows } = timeline;
+
+        for (const row of rows) {
+          const { layers } = row;
+
+          for (const layer of layers) {
+            const { filter } = layer;
+
+            if (filter.activity !== undefined) {
+              const { activity: activityFilter } = filter;
+              const { types } = activityFilter;
+              const includeActivity = types.indexOf(activityDirective.type) > -1;
+
+              if (includeActivity) {
+                if (byLayerId[layer.id] === undefined) {
+                  byLayerId[layer.id] = [activityDirective];
+                } else {
+                  byLayerId[layer.id].push(activityDirective);
+                }
+
+                if (byTimelineId[timeline.id] === undefined) {
+                  byTimelineId[timeline.id] = [activityDirective];
+                } else {
+                  byTimelineId[timeline.id].push(activityDirective);
+                }
               }
             }
           }
         }
       }
     }
-  }
 
-  return { byLayerId, byTimelineId };
-});
+    return { byLayerId, byTimelineId };
+  },
+);
 
-export const allActivityTags: Readable<string[]> = derived(activities, $activities => {
-  const tagMap = $activities.reduce((map: Record<string, boolean>, activity: Activity) => {
-    activity.tags.forEach(tag => (map[tag] = true));
-    return map;
-  }, {});
+export const allActivityDirectiveTags: Readable<string[]> = derived(activityDirectivesList, $activityDirectivesList => {
+  const tagMap = $activityDirectivesList.reduce(
+    (map: Record<string, boolean>, activityDirective: ActivityDirective) => {
+      activityDirective.tags.forEach(tag => (map[tag] = true));
+      return map;
+    },
+    {},
+  );
   return Object.keys(tagMap).sort();
 });
 
-export const selectedActivity = derived(
-  [activitiesMap, selectedActivityId],
-  ([$activitiesMap, $selectedActivityId]) => $activitiesMap[$selectedActivityId] || null,
+export const selectedActivityDirective = derived(
+  [activityDirectivesMap, selectedActivityDirectiveId],
+  ([$activityDirectivesMap, $selectedActivityDirectiveId]) =>
+    $activityDirectivesMap[$selectedActivityDirectiveId] || null,
 );
 
 /* Helper Functions. */
 
+export function selectActivity(
```

Comment:
One potential quirk here is if you've closed/shrunk the activities table and then select an activity from the timeline, the table re-opens if it's closed and then reverts to the default height. I was imagining the table would stay at its existing height if open and if closed then maybe stay closed?